src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java

```diff
@@ -29,25 +19,23 @@
   private static final long DEFAULT_TIMEOUT = 30000;
 
   /**
-   * Creates a {@link BlockingSphereClient} with a custom {@code timeout} with a custom {@link
-   * TimeUnit}.
+   * Creates a {@link SphereClient} with a default {@code timeout} value of 30 seconds.
    *
    * @param clientConfig the client configuration for the client.
-   * @return the instantiated {@link BlockingSphereClient}.
+   * @return the instantiated {@link SphereClient}.
    */
   public static SphereClient createClient(@Nonnull final SphereClientConfig clientConfig) {
 
     final HttpClient httpClient = getHttpClient();
-    final SphereAccessTokenSupplier tokenSupplier =
-        SphereAccessTokenSupplier.ofAutoRefresh(clientConfig, httpClient, false);
-    final SphereClient underlying = SphereClient.of(clientConfig, httpClient, tokenSupplier);
-    final SphereClient retryClient = withRetry(underlying);
-    return withLimitedParallelRequests(retryClient);
+    return RetryableSphereClientBuilder.of(clientConfig, httpClient)
+        .withStatusCodesToRetry(
+            Arrays.asList(BAD_GATEWAY_502, SERVICE_UNAVAILABLE_503, GATEWAY_TIMEOUT_504))
```

Comment:
I think the default is better, which includes 500 too.

Reply:
c61c772